motionplan/ensembleIKSolver.go

```diff
@@ -12,48 +12,36 @@ import (
 	"go.viam.com/rdk/spatialmath"
 )
 
-// CombinedIK defines the fields necessary to run a combined solver.
-type CombinedIK struct {
-	solvers []InverseKinematics
-	model   referenceframe.Frame
-	logger  golog.Logger
+// ensembleIKSolver defines the fields necessary to run a combined solver.
+type ensembleIKSolver struct {
+	*ikSolver
+	solvers []inverseKinematicsSolver
 }
 
-// CreateCombinedIKSolver creates a combined parallel IK solver with a number of nlopt solvers equal to the nCPU
+// newEnsembleIKSolver creates a combined parallel IK solver with a number of nlopt solvers equal to the nCPU
 // passed in. Each will be given a different random seed. When asked to solve, all solvers will be run in parallel
 // and the first valid found solution will be returned.
-func CreateCombinedIKSolver(model referenceframe.Frame, logger golog.Logger, nCPU int) (*CombinedIK, error) {
-	ik := &CombinedIK{}
-	ik.model = model
-	if nCPU == 0 {
-		nCPU = 1
-	}
-	for i := 1; i <= nCPU; i++ {
-		nlopt, err := CreateNloptIKSolver(model, logger, -1)
+func newEnsembleIKSolver(model referenceframe.Frame, logger golog.Logger, opts *ikOptions) (*ensembleIKSolver, error) {
+	ik := &ensembleIKSolver{ikSolver: &ikSolver{
+		logger: logger,
+		model:  model,
+		opts:   opts,
+	}}
+
+	for i := 1; i <= opts.NumThreads; i++ {
+		nlopt, err := newNLOptIKSolver(model, logger, opts)
```

Comment:
Related to Peter's suggestion, you could specify the seed for each new NLOptIKSolver here if you had a "starting seed" as part of the `opts` struct.

Reply:
It's tough to build it into the solver struct itself, because the solver struct ought generally be thread-safe, while rand seeds are not

Reply:
OK, fair enough